source/host_resolver.c

```diff
@@ -1020,9 +1050,21 @@ static void resolver_thread_fn(void *arg) {
                 host_entry->host_name);
             aws_mutex_unlock(&host_entry->entry_lock);
 
-            AWS_ASSERT(err_code != AWS_ERROR_SUCCESS || aws_array_list_length(&callback_address_list) > 0);
+            size_t callback_list_size = aws_array_list_length(&callback_address_list);
+            if (callback_list_size > 0) {
+                AWS_LOGF_DEBUG(
+                    AWS_LS_IO_DNS,
+                    "static, invoking resolution callback for host %s with %d addresses",
+                    aws_string_c_str(host_entry->host_name),
+                    (int)callback_list_size);
+            } else {
+                AWS_LOGF_DEBUG(
```

Comment:
This log message doesn't seem accurate to what is actually happening? Isn't this just "there were no callbacks registered"?

Reply:
No, this is there's no addresses available